### ISSUE TITLE
fetchMultiple should return strongly typed results instead of Object | null

### ISSUE THREAD

**Issue by akutruff** [coral-xyz/anchor#1465]
It would be better if the results of `fetchMultiple` were strongly typed.
`const games = await program.account.gameAccount.fetchMultiple(['address0', 'address1]);` 

`fetchMultiple` has a return type of `Promise<(Object | null)[]>` rather than `Promise<(TypeDef<{  /*etc*/  }  | null)[]>`

**Comment by akutruff** [comment]
This is widening into a broader issue.  It appears that even `fetch()` returns an incorrect type for an account.  The returned type is actually the union of all possible properties on all account types defined in Rust.

`type GameAccount = Awaited<ReturnType<Program<MyProgram>['account']['gameAccount']['fetch']>>;`  

The above type has all the properties on all types in the program, not just the ones for the `GameAccount` type in Rust.

In general, the current approach for type generation spits out a TypeScript type which is more like metadata for the fluent interface of `Program<T>`  As users, we're looking for a 1:1 mapping of the Rust types to the TypeScript types.  It may be better to just do the fully resolved type generation at build time so that all the accounts have fully defined type names.  Of course, I don't know all the nuances that lead to the current approach, and it may be the best option.  However, I figured I'd throw this out there. 

In other words, have the generated `.ts` file have something like:

```
interface GameAccount {
   id: string;
   playerCount: number;
}
```
etc.   rather than the current type definitions.
